test/unit-test/cellular_at_core_utest.c

```diff
@@ -0,0 +1,932 @@
+    cellularStatus = Cellular_ATGetSpecificNextTok( & pString, pDelimiter, & pTokOutput );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+    free( pStringSource );
+}
+
+/**
+ * @brief Test the error path for Cellular_ATGetSpecificNextTok to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATGetSpecificNextTok_Error_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString = "";
+    const char * pDelimiter = ",";
+    char * pTokOutput;
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATGetSpecificNextTok( & pString, pDelimiter, & pTokOutput );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test that any NULL parameter causes Cellular_ATHexStrToHex to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATHexStrToHex_Invalid_Param( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString;
+    uint8_t * const * const ppHexData;
+    uint16_t hexDataLen;
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATHexStrToHex( NULL, ppHexData, hexDataLen );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATHexStrToHex( pString, NULL, hexDataLen );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATHexStrToHex to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATHexStrToHex_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char pString[] = CELLULAR_SAMPLE_HEX_STRING;
+    char pLowerCaseString[] = CELLULAR_SAMPLE_HEX_LOWER_CASE_STRING;
+    uint8_t * ppHexData[3];
+    uint16_t hexDataLen = 3;
+
+    /* Check that CELLULAR_AT_SUCCESS is returned. */
+    cellularStatus = Cellular_ATHexStrToHex( pString, ppHexData, hexDataLen );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );    
+
+    /* Check that CELLULAR_AT_SUCCESS is returned. */
+    cellularStatus = Cellular_ATHexStrToHex( pLowerCaseString, ppHexData, hexDataLen );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );    
+}
+
+/**
+ * @brief Test the error path for Cellular_ATHexStrToHex to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATHexStrToHex_Error_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString = "";
+    uint8_t * const * const ppHexData;
+    uint16_t hexDataLen = 0;
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATHexStrToHex( pString, ppHexData, hexDataLen );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    pString = CELLULAR_SAMPLE_HEX_STRING;
+    hexDataLen = 2;
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */

```

Comment:
The comment says that check CELLULAR_BAD_PARAMETER is returned while assert checks CELLULAR_AT_NO_MEMORY. Also, this can be broken in 2 test cases. The idea is to just test one thing in one test case.